evennia/objects/objects.py

```diff
@@ -612,25 +612,56 @@ def msg(self, text=None, from_obj=None, session=None, options=None, **kwargs):
                 (which can be a string or a tuple).
 
         Notes:
-            `at_msg_receive` will be called on this Object.
+            `at_msg_receive`and at_post_msg_receive will be called on this Object.
             All extra kwargs will be passed on to the protocol.
 
         """
-        # try send hooks
-        if from_obj:
-            for obj in make_iter(from_obj):
-                try:
-                    obj.at_msg_send(text=text, to_obj=self, **kwargs)
-                except Exception:
-                    logger.log_trace()
         kwargs["options"] = options
-        try:
-            if not self.at_msg_receive(text=text, from_obj=from_obj, **kwargs):
-                # if at_msg_receive returns false, we abort message to this object
-                return
-        except Exception:
-            logger.log_trace()
+        self._msg_helper_text_format(text, kwargs)
+
+        # try send hooks
+        self._msg_helper_from_obj(from_obj=from_obj, **kwargs)
```

Comment:
While I can see the point of these being private from a code perspective, using a private `_` prefix also suggests that this is a private method that is not intended to be customized by the developer.